openmdao/approximation_schemes/approximation_scheme.py

```diff
@@ -77,26 +110,457 @@ def compute_approximations(self, system, jac=None, total=False):
+                                            result[out_slices[of]][out_idxs]).copy()))
+                    else:
+                        J['data'][:, i_count] = self._transform_result(result[full_idxs])
+
+                fd_count += 1
+
+        mult = self._get_multiplier(data)
+        if colored_shape is not None:  # coloring is active
+            if par_fd_w_serial_model:
+                if self._j_colored is None:
+                    jstuff = mycomm.allgather((jrows, jcols, jdata))
+                    rowlist = [rows for rows, _, _ in jstuff if rows]
+                    allrows = np.hstack(rowlist)
+                    allcols = np.hstack(cols for _, cols, _ in jstuff if cols)
+                    alldata = np.hstack(dat for _, _, dat in jstuff if dat)
+                    self._j_colored = coo_matrix((alldata, (allrows, allcols)), shape=colored_shape)
+                    self._j_data_sizes = sizes = np.array([len(x) for x, _, _ in jstuff])
+                    self._j_data_offsets = offsets = np.zeros(mycomm.size)
+                    offsets[1:] = np.cumsum(sizes)[:-1]
+                else:
+                    mycomm.Allgatherv(jdata, [self._j_colored.data, self._j_data_sizes,
+                                              self._j_data_offsets, MPI.DOUBLE])
+
+            elif is_parallel:
+                raise NotImplementedError("colored FD/CS over parallel groups not supported yet")
+            else:  # serial colored
+                if do_rows_cols:
+                    self._j_colored = coo_matrix((jdata, (jrows, jcols)), shape=colored_shape)
+                else:
+                    self._j_colored.data[:] = jdata
+
+            if mult != 1.0:
+                self._j_colored.data *= mult
+
+            # convert COO matrix to dense for easier slicing
+            Jcolored = self._j_colored.toarray()
+
+        elif is_parallel:  # uncolored with parallel systems
+            results = _gather_jac_results(mycomm, results)
+
+        for _, _, tmpJ, _, _ in colored_approx_groups:
+            # TODO: coloring when using parallel FD and/or FD with remote comps
+            for key in tmpJ['@approxs']:
+                slc = tmpJ['@jac_slices'][key]
+                if uses_voi_indices:
+                    jac._override_checks = True
+                    jac[key] = _from_dense(jacobian, key, Jcolored[slc])
+                    jac._override_checks = False
+                else:
+                    jac[key] = _from_dense(jacobian, key, Jcolored[slc])
+
+        Jcolored = None  # clean up memory
+
+        for wrt, _, _, tmpJ, _, _ in approx_groups:
+            ofs = tmpJ[wrt]['ofs']
+            for of in ofs:
+                key = (of, wrt)
+                oview, _, rows_reduced, cols_reduced = ofs[of]
+                if is_parallel:
+                    for i, result in results[key]:
+                        oview[:, i] = result
+
+                oview *= mult
```

Comment:
Perhaps you should put a `if mult != 1.0:` around this too.

Reply:
done